docs/components/checkbox/_spec.scss

```diff
@@ -0,0 +1,113 @@
+@import "components/checkbox/config";
+
+.checkbox-spec {
+  hx-checkbox-control {
+    > label {
+      > hx-checkbox {
+        display: none;
+      }
+    }
+
+    /* ----- PRISTINE ----- */
+    > input[type="checkbox"] {
+      /* imitate :focus */
+      &.mock-focus {
+        // &:extend(hx-checkbox-control > input[type="checkbox"]:focus all);
+        @include hxCheckboxControl($psuedo-state: focus);
+      }
+
+      /* imitate :hover */
+      &.mock-hover {
+        // &:extend(hx-checkbox-control > input[type="checkbox"]:hover all);
+        @include hxCheckboxControl($psuedo-state: hover);
+
+        ~ label > hx-checkbox {
+          // &:extend(hx-checkbox-control > input[type="checkbox"]:valid ~ label > hx-checkbox:hover all);
+          @include hxCheckboxFacade($psuedo-state: hover);
+        }
+      }
+
+      /* imitate :disabled:focus */
+      &:disabled.mock-focus {
+        // &:extend(hx-checkbox-control > input[type="checkbox"]:disabled:focus all);
+        @include hxCheckboxControl(disabled, focus);
+      }
+
+      /* imitate :disabled:hover */
+      &:disabled.mock-hover {
+        // &:extend(hx-checkbox-control > input[type="checkbox"]:disabled:hover all);
+        @include hxCheckboxControl(disabled, hover);
+
+        ~ label > hx-checkbox {
+          // &:extend(hx-checkbox-control > input[type="checkbox"]:disabled ~ label > hx-checkbox:hover all);
+          @include hxCheckboxFacade(disabled, hover);
+        }
+      }
+    }
+
+    /* ----- INVALID ----- */
+    &[hx-dirty] {
+      > input[type="checkbox"]:invalid {
+        /* imitate :focus */
+        &.mock-focus {
+          // &:extend(hx-checkbox-control[hx-dirty] > input[type="checkbox"]:invalid:focus all);
+          @include hxCheckboxControl(invalid, focus);
+        }
+
+        /* imitate :hover */
+        &.mock-hover {
+          // &:extend(hx-checkbox-control[hx-dirty] > input[type="checkbox"]:invalid:hover all);
+          @include hxCheckboxControl(invalid, hover);
+
+          ~ label > hx-checkbox {
+            // &:extend(hx-checkbox-control[hx-dirty] > input[type="checkbox"]:invalid ~ label > hx-checkbox:hover all);
+            @include hxCheckboxFacade(invalid, hover);
+          }
+        }
+
+        /* default invalid styles */
+        ~ label > hx-checkbox {
+          @include hxCheckboxFacade($state: invalid);
+        }
+      }
+    }
+
+    &.hxInvalid {
+      > input[type="checkbox"]:enabled {
+        /* imitate :focus */
+        &.mock-focus {
+          // &:extend(hx-checkbox-control.hxInvalid > input[type="checkbox"]:enabled:focus all);
```

Comment:
❗️ I keep the old LESS &:extends (commented out) to show what I changed.  I'll remove before merging.